spec/controllers/work_groups_controller_spec.rb

```diff
@@ -1,19 +1,118 @@
 require 'rails_helper'
 
 RSpec.describe WorkGroupsController, type: :controller do
+  let(:teacher)     { create(:teacher) }
+  let(:student1)    { create(:student) }
+  let(:student2)    { create(:student) }
+  let(:classroom)   { create(:classroom, user: teacher) }
+  let(:work_group1) { create(:work_group, classroom: classroom) }
+  let(:worksheet_template) { create(:worksheet_template, user: teacher) }
+  # let(:worksheet_template2) { create(:worksheet_template, image_url: 'google.com', user: teacher) }
+  let(:worksheet1) { create(:worksheet, work_group: work_group1, worksheet_template: worksheet_template) }
 
-  describe "GET #new" do
-    it "returns http success" do
+  before do
+    request.headers['X-User-Email'] = teacher.email
+    request.headers['X-User-Token'] = teacher.authentication_token
+  end
+
+  describe 'GET #new' do
+    context 'success' do
+      it 'assigns correct new work group with params' do
+        worksheet1
+        put :new, params: {
+          work_group: { turn_time: 300_000 },
+          no_model_fields: {
+            emails: 'test1@gmail.com test2@gmail.com',
+            worksheet_url: 'https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg',
+            start_time: DateTime.now + 1.day,
+            start_date: Date.today
+          }
+        }
+        expect(controller.instance_variable_get(:@work_group).turn_time).to eq(300_000)
+        expect(WorkGroup.find_by_turn_time(300_000)).to be_instance_of(WorkGroup)
+        expect(controller.instance_variable_get(:@worksheet_urls)).to eq(['https://res.cloudinary.com/naokimi/image/upload/v1563422680/p7ojmgdtwshkrhxmjzh1.jpg'])
```

Comment:
PROBLEM! So I'm not even sure if this is a necessary test, but I have a problem where the WorksheetTemplate isn't owned by a user with id=1. So this always returns an empty array. I thought it was because I was abusing the let statements, but I tried commenting that out and it still had the same problem. To my understanding, when I call the 'new' method it creates a new WorkGroup object which also creates a new User. So I thought that the new User would have an id=1 being the first created object that is stored (or maybe it's not stored....) in the db. Hmmmm....

Reply:
to make sure that the worksheet template is owned by user with ID 1, you need to make sure that the teacher in https://github.com/odaiba-app/server/pull/140/files#diff-6b9cb644ea03481672b7f5bc690e9c7068cdc8dd5bd3f84163d6c337c592d089R5 is created with an ID of 1
try adding a `binding.pry` and see if calling `teacher` at line 22 will create him with the right ID